services/core-api/app/api/parties/party/models/address.py

```diff
@@ -87,17 +87,15 @@ def create(cls,
         return address
 
     # will be called for both fields, in the order defined in model, 1st call will be missing 2nd value
-    @validates('post_code', 'address_type_code')
-    def validate_address_code(self, key, value):
-        if key == 'address_type_code' and self.post_code is not None:
-            maxLength = 6
-            validPostalCode = re.compile(r"(^[ABCEGHJ-NPRSTVXY]\d[ABCEGHJ-NPRSTV-Z]\d[ABCEGHJ-NPRSTV-Z]\d$)")
-            if value == 'USA':
-                maxLength = 10
-                validPostalCode = re.compile(r"((^\d{5}$)|(^\d{9}$)|(^\d{5}-\d{4}$))")
-            if len(self.post_code) > maxLength:
-                raise AssertionError(f'post_code must not exceed {maxLength} characters.')
-            if not validPostalCode.match(self.post_code):
-                current_app.logger.error(f'Failed post_code validation for address {self.address_id}. post_code: {self.post_code}, address_type_code: {self.address_type_code}, value: {value}',)
-                raise AssertionError(f'Invalid post_code format.')
-        return value
+    @validates('post_code')
+    def validate_address_code(self, key, post_code):
+        if post_code and len(post_code) > 10:
+            raise AssertionError('post_code must not exceed 6 characters.')
```

Comment:
Since we're just checking the length once for both Canadian and US, maybe we should make the error more generic.

Something like "post_code exceeds the maximum allowed characters"?